driver-core/src/main/com/mongodb/MongoOperationTimeoutException.java

```diff
@@ -35,6 +37,7 @@
  * @see MongoClientSettings#getTimeout(TimeUnit)
  * @since CSOT
  */
+@Alpha(Alpha.Reason.CLIENT)
```

Comment:
It seems to me that if `MongoOperationTimeoutException` is `@Alpha`, we should not throw it unless CSOT is specified. However, `SocketStreamHelper.initialize` calls `TimeoutContext.createConnectTimeoutMs`, which throws `MongoOperationTimeoutException` when `TimeoutContext` expires, regardless of whether `TimeoutContext.timeout` is `null` or not. I.e., at the very least `SocketStreamHelper.initialize`, but maybe also some other parts of the driver, throw `MongoOperationTimeoutException` even if CSOT is not specified.

Reply:
The MongoOperationTimeoutException should indeed only be thrown when CSOT is enabled. This appears to be a newly introduced bug following the Timeout API refactoring. The exception should not be thrown here, as there is  code a few layers up specifically used to manage SocketTimeoutException and decide which exception should be relayed to the user. 

I will investigate this issue further and check that MongoOperationTimeout exception is not exposed to the user when CSOT is enabled in a separate ticket.  However, i would leave `MongoOperationTimeoutException` marked as @Alpha for now.

Good catch! Thanks!

Reply:
The `MongoOperationTimeoutException` should indeed only be thrown when CSOT is enabled. This appears to be a newly introduced bug following the Timeout API refactoring. I will investigate this issue further in a separate ticket JAVA-5439. However, the `MongoOperationTimeoutException` should remain marked as @Alpha.